app/controllers/topics_controller.rb

```
@@ -94,6 +95,26 @@ def show
+    tu = TopicUser.find_by(user_id: current_user.id, topic_id: params[:topic_id])
+
+    if tu.notification_level > TopicUser.notification_levels[:regular]
+      tu.notification_level = TopicUser.notification_levels[:regular]
+    else
+      tu.notification_level = TopicUser.notification_levels[:muted]
+    end
+
+    tu.save!
+
+    perform_show_response
+  end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `unsubscribe` is too high. [<4, 31, 6> 31.83/23]

(Metrics/AbcSize)